### DISCUSSION TITLE
AR viewing works on iOS but not Android

### DISCUSSION THREAD

**Discussion by ferb7nft** [google/model-viewer#2588] [Q&A] [answered]
I have followed all of the tutorials online, but the AR button and view only seems to work on iOS devices and not Android. I have tried multiple iterations of code, but will share one here that is currently allowing AR on iOS but not Android. Also, I have tried other websites on Android devices that successfully deploy AR, so I know the devices are compatible. Here is my code that will not deploy AR view on Android, but can on iOS: `<model-viewer
      src="file.glb"
      ios-src="file.usdz"
      ar-modes="webxr scene-viewer quick-look"
      camera-controls
      loading="eager"
      auto-rotate
      auto-rotate-delay="0"
      shadow-intensity="0"
      ar >`

model-viewer documentation says "If Google Play Services with AR isn't present, tapping this button displays the model in <model-viewer>'s 3D viewer." Although, I tried the demos on the model-viewer site in android, and they worked! And I checked, and the device has "Google Play Services with AR" installed! So I'm not sure why my code above isn't working for my site...

**Comment by elalish** [comment]
My best guess is permissions; WebXR will ask for AR permission once per URL, and if you block it, it will silently fail from then on unless you manually go in and clear the permissions so it can ask again. You can also switch the order of "webxr scene-viewer" to get the scene-viewer experience instead. Also, what browser/device are you using?

**Reply by ferb7nft** [reply]
Ok I will add scene-viewer first and try and clear the browser data. I tried on Google Chrome and Samsung Internet for Android.

**Reply by ferb7nft** [reply]
Here is the website and code:

https://ipfs.io/ipfs/QmPLG92kQuXjGVuv74gbuXpfoxiBAuzmmfAaNGe9cBcZt4/

[index.txt](https://github.com/google/model-viewer/files/6842492/index.txt)

**Reply by ferb7nft** [reply]
And this website and code also does not work

**Comment by ferb7nft** [comment, marked as answer]
The problem was an outdated api version. It was fixed by using <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
